packages/block-library/src/file/save.js

```diff
@@ -13,11 +14,14 @@ export default function save( { attributes } ) {
 		downloadButtonText,
 	} = attributes;
 
+	const fileId = `wp-block-file--media-${ id }`;
+
 	return (
 		href && (
 			<div>
 				{ ! RichText.isEmpty( fileName ) && (
 					<a
+						id={ fileId }
```

Comment:
It's difficult to make this id absolutely unique. I think an issue here is that the same file could be inserted twice into a post.

I made an issue a while ago about how difficult it is to create a consistent id within a block:
https://github.com/WordPress/gutenberg/issues/17246

Reply:
> It's difficult to make this id absolutely unique. I think an issue here is that the same file could be inserted twice into a post.

@talldan Is this an issue that needs to be resolved for this PR? Or a separate issue (#17246) that needs to be resolved in general?

Reply:
It's an issue for this PR, in my opinion. Using the block `clientId` could be an option. Some care would have to be taken to ensure it's updated when a block is duplicated.

Or another option might be to avoid `aria-describedby` entirely and use `aria-description` instead.

Reply:
@talldan perhaps id could be a hash of the file? I mean if it's good enough for Git, I'm not sure most websites will break it.

Reply:
> Or another option might be to avoid `aria-describedby` entirely and use `aria-description` instead.

`aria-description` works but is not considered a valid ARIA attribute by [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/aria-props.md).

Reply:
I've opened #28062 to make these changes but use `clientId` as a unique id.